docs/threading.md

```diff
@@ -44,31 +46,20 @@ steps necessary. If you are using a MinGW or similar environment where
 libssh2 uses OpenSSL or libgcrypt, then the general case affects
 you.
 
-On Mac OS X
+On macOS
 -----------
 
-By default we use libcurl to perform the encryption. The
-system-provided libcurl uses SecureTransport, so no special steps are
-necessary. If you link against another libcurl (e.g. from homebrew)
-refer to the general case.
-
-If the option to use libcurl was deactivated, the library makes use of
-CommonCrypto and SecureTransport for cryptographic support. These are
-thread-safe and you do not need to do anything special.
+By default we make use of CommonCrypto and SecureTransport for cryptographic
+support. These are thread-safe and you do not need to do anything special.
 
 Note that libssh2 may still use OpenSSL itself. In that case, the
 general case still affects you if you use ssh.
 
 General Case
 ------------
 
-If it's available, by default we use libcurl to provide HTTP tunneling support,
-which may be linked against a number of cryptographic libraries and has its
-own
-[recommendations for thread safety](https://curl.haxx.se/libcurl/c/threadsafe.html).
-
-If there are no alternative TLS implementations (currently only
-SecureTransport), libgit2 uses OpenSSL in order to use HTTPS as a transport.
+If there are no applicable TLS implementations (currently only
+SecureTransport and mbedTLS), libgit2 defaults to OpenSSL in order to use HTTPS as a transport.
```

Comment:
"applicable" feels like the wrong word here (OpenSSL would also be "applicable"), which I think is why it says "alternative" (these are the "alternatives" to the default of OpenSSL).

I think we should just be more clear about the defaults, something like:

> By default, libgit2 uses the Secure Transport on macOS, WinHTTP and CryptoNG on Windows and OpenSSL on all other systems.  mbedTLS may be optionally enabled as the security provider.